test/e2e/windows/memory_limits.go

```diff
@@ -101,64 +99,23 @@ func checkNodeAllocatableTest(f *framework.Framework) {
 // Deploys `allocatablePods + 1` pods, each with a memory limit of `1/allocatablePods` of the total allocatable
 // memory, then confirms that the last pod failed because of failedScheduling
 func overrideAllocatableMemoryTest(f *framework.Framework, allocatablePods int) {
-	const (
-		podType = "memory_limit_test_pod"
-	)
 
-	totalAllocatable := getTotalAllocatableMemory(f)
-
-	memValue := totalAllocatable.Value()
-	memPerPod := memValue / int64(allocatablePods)
-	ginkgo.By(fmt.Sprintf("Deploying %d pods with mem limit %v, then one additional pod", allocatablePods, memPerPod))
-
-	// these should all work
-	pods := newMemLimitTestPods(allocatablePods, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().CreateBatch(pods)
-
-	failurePods := newMemLimitTestPods(1, imageutils.GetPauseImageName(), podType, strconv.FormatInt(memPerPod, 10))
-	f.PodClient().Create(failurePods[0])
-
-	gomega.Eventually(func() bool {
-		eventList, err := f.ClientSet.CoreV1().Events(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
-		framework.ExpectNoError(err)
-		for _, e := range eventList.Items {
-			// Look for an event that shows FailedScheduling
-			if e.Type == "Warning" && e.Reason == "FailedScheduling" && e.InvolvedObject.Name == failurePods[0].ObjectMeta.Name {
-				framework.Logf("Found %+v event with message %+v", e.Reason, e.Message)
-				return true
-			}
-		}
-		return false
-	}, 3*time.Minute, 10*time.Second).Should(gomega.Equal(true))
-}
-
-// newMemLimitTestPods creates a list of pods (specification) for test.
-func newMemLimitTestPods(numPods int, imageName, podType string, memoryLimit string) []*v1.Pod {
-	var pods []*v1.Pod
-
-	memLimitQuantity, err := resource.ParseQuantity(memoryLimit)
+	selector := labels.Set{"kubernetes.io/os": "windows"}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
 	framework.ExpectNoError(err)
 
-	for i := 0; i < numPods; i++ {
-
-		podName := "test-" + string(uuid.NewUUID())
-		pod := v1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: podName,
-				Labels: map[string]string{
-					"type": podType,
-					"name": podName,
-				},
-			},
+	for _, node := range nodeList.Items {
+		status := node.Status
+		pod := &v1.Pod{
 			Spec: v1.PodSpec{
```

Comment:
this pod spec should also include a `node: node.Name` field to make it deploy to that node